rpc/client.go

```diff
@@ -187,22 +190,27 @@ func (c *Client) Healthy() <-chan struct{} {
 // Close closes the Closed channel, which triggers the end of the run loop and
 // removal from the clients map.
 func (c *Client) Close() {
+	c.close()
+	<-c.Closed
+}
+
+func (c *Client) close() {
```

Comment:
updated the comments, though the "third" channel passed in via retryOpts is not going to be documented here since it's part of `retry`'s surface, not the client's.